### PR TITLE
Create PDA hash table example

### PR DESCRIPTION
**Pull request by LucasSte**
I saw an example of a PDA hash table on [Anchor's website](https://www.anchor-lang.com/docs/pdas#hashmap-like-structures-using-pd-as). I liked it and wanted to replicate it in Solidity.

### CODE REVIEW

**Review by xermicus — APPROVED** [review]
I'd reference the anchor docs somewhere to credit the source and provide more info behind it

File: integration/solana/setup.ts [hunk 1]
```diff
@@ -6,30 +6,35 @@ import fs from 'fs';
...
-export async function loadContract(name: string, args: any[] = [], space: number = 8192):
+export async function loadContractAndCallConstructor(name: string, args: any[] = [], space: number = 8192):
     Promise<{ program: Program, payer: Keypair, provider: AnchorProvider, storage: Keypair, program_key: PublicKey }> {
 
-    const idl = JSON.parse(fs.readFileSync(`${name}.json`, 'utf8'));
-
-    const payer = loadKey('payer.key');
+    const {program, payer, provider, program_key} = await loadContract(name);
 
     process.env['ANCHOR_WALLET'] = 'payer.key';
```
Comment (seanyoung) [inline integration/solana/setup.ts:14]: `loadContract()` already does this

**Review by seanyoung — APPROVED** [review]

File: integration/solana/setup.ts (hunk 1 above)
Comment (LucasSte) [reply integration/solana/setup.ts:14]: Yes, you're right. That slipped through.

**Comment by LucasSte** [comment]
> I'd reference the anchor docs somewhere to credit the source and provide more info behind it

I included a reference to Anchor's website and said that our example was inspired by theirs.

**Pull request merged**